### ISSUE TITLE
"Monthly Active User Limit Exceeded" notice sent twice

### ISSUE THREAD

**Issue by Half-Shot** [matrix-org/synapse#9469]
![image](https://user-images.githubusercontent.com/2072976/108832303-457fad80-75c3-11eb-8ef5-5161e64e7d3e.png)

The logging is fairly light on details:
```
2021-02-23 10:35:27,689 - synapse.server_notices.server_notices_manager - 81 - INFO - GET-137 - Sending server notice to @will:beefy
2021-02-23 10:35:27,706 - synapse.handlers.message - 921 - INFO - GET-139 - Not bothering to persist state event $fAoSuk8bwtNq7-K9MOnZrS7NgZh52-EeWWH6NwjYIS0 duplicated by $WzlsT3cfrTglrbzX0cqRCWh5KvlTYPu0npEgsYn5uQY
2021-02-23 10:35:27,706 - synapse.server_notices.server_notices_manager - 81 - INFO - GET-139 - Sending server notice to @will:beefy
2021-02-23 10:35:27,717 - synapse.server_notices.server_notices_manager - 81 - INFO - GET-137 - Sending server notice to @will:beefy
2021-02-23 10:35:27,736 - synapse.server_notices.server_notices_manager - 81 - INFO - GET-139 - Sending server notice to @will:beefy
```

I suspect there is a race in the code

**Comment by clokep** [comment]
> I suspect there is a race in the code

What version of Synapse is this? Is this a setup with workers?

**Comment by Half-Shot** [comment]
Synapse 1.28-rc1, no workers.

**Comment by clokep** [comment]
> Synapse 1.28-rc1, no workers.

Just to check -- was this working correctly in previous versions (is it a regression?) or do you think it is a bug that has always existed?

**Comment by Half-Shot** [comment]
I've never tested this before, but I just noticed this while testing today so I'm afraid I'm not sure.

**Comment by Half-Shot** [comment]
I think this might be fixed by #9544. I'm going to close it, and we can reopen if we see it.
